core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -256,13 +258,22 @@ class KafkaServer(
         tokenCache = new DelegationTokenCache(ScramMechanism.mechanismNames)
         credentialProvider = new CredentialProvider(ScramMechanism.mechanismNames, tokenCache)
 
+        /* start forwarding manager */
         if (enableForwarding) {
+          clientToControllerChannelManager =
+            Some(
+              BrokerToControllerChannelManager(
+                controllerNodeProvider = MetadataCacheControllerNodeProvider(config, metadataCache),
+                time = time,
+                metrics = metrics,
+                config = config,
+                channelName = "clientToControllerChannel",
+                threadNamePrefix = threadNamePrefix,
+                retryTimeoutMs = config.requestTimeoutMs.longValue)
+            )
+          clientToControllerChannelManager.get.start()
           this.forwardingManager = Some(ForwardingManager(
-            config,
-            metadataCache,
-            time,
-            metrics,
-            threadNamePrefix
+            clientToControllerChannelManager.get
```

Comment:
nit: this `get` here is a little ugly. We could create a local `val` when we construct `BrokerToControllerChannelManager`.